app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java

```diff
@@ -755,6 +758,11 @@ public boolean onPreferenceChange(Preference preference, Object value) {
             String stringValue = value.toString();
             if (isNumeric(stringValue)) {
                 final boolean domgdl = Pref.getString("units", "mgdl").equals("mgdl"); // Identify which unit is chosen
+                double submissionMgdl = domgdl ? Double.parseDouble(stringValue) : Double.parseDouble(stringValue) * Constants.MMOLL_TO_MGDL;
```

Comment:
tolerantParseDouble() is typically much safer to use and supports localized number input where , can be the decimal point.

Reply:
Thanks
Will change it